opm/autodiff/MultisegmentWell_impl.hpp

```diff
@@ -868,15 +868,14 @@ namespace Opm
     template <typename TypeTag>
     typename MultisegmentWell<TypeTag>::EvalWell
     MultisegmentWell<TypeTag>::
-    volumeFraction(const int seg, const int comp_idx) const
+    volumeFraction(const int seg, const unsigned compIdx) const
     {
-        const PhaseUsage& pu = phaseUsage();
 
-        if (active()[Water] && comp_idx == pu.phase_pos[Water]) {
+        if (active()[Water] && compIdx == Indices::canonicalToActiveComponentIndex(FluidSystem::waterCompIdx)) {
             return primary_variables_evaluation_[seg][WFrac];
         }
 
-        if (active()[Gas] && comp_idx == pu.phase_pos[Gas]) {
+        if (active()[Gas] && compIdx == Indices::canonicalToActiveComponentIndex(FluidSystem::gasCompIdx)) {
```

Comment:
Should all the `active()` and `active_` be gone if following `BlackoilModeEbos`?

Reply:
It is on my todo list, but you are right, it belongs to this PR.